probe-rs/src/architecture/arm/core/armv7a.rs

```diff
@@ -279,7 +279,7 @@ impl<'probe> Armv7a<'probe> {
         &mut self,
         op: impl FnOnce(&mut Self) -> Result<R, Error>,
     ) -> Result<R, Error> {
-        let was_running = !(self.state.current_state.is_halted() || self.core_halted()?);
+        let was_running = !self.state.current_state.is_halted();
```

Comment:
This change is unrelated to this PR and likely not correct. A running core can halt for any reason, so in case we cached `is_halted() == false` that may not be the real state of the core.

Reply:
For whatever reason, I was getting errors without this change. Something is desynchronizing it sometimes. There are cases where the core is halted but the current state is not halted.

Perhaps it has to do with resetting the core, which happens out of band.

Either way, I'll drop the patch from this PR and try to determine why it's necessary.